build-id-versioning/workflows.go

```diff
@@ -0,0 +1,134 @@
+package build_id_versioning
+
+import (
+	"context"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+// SampleChangingWorkflowV1 is a workflow we'll be making changes to, and represents the first
+// version
+func SampleChangingWorkflowV1(ctx workflow.Context) error {
+	workflow.GetLogger(ctx).Info("Changing workflow v1 started.", "StartTime", workflow.Now(ctx))
+
+	// This workflow will listen for signals from our starter, and upon each signal either run
+	// an activity, or conclude execution.
+	signalChan := workflow.GetSignalChannel(ctx, "do-next-signal")
+	for {
+		var signal string
+		signalChan.Receive(ctx, &signal)
+
+		if signal == "do-activity" {
+			workflow.GetLogger(ctx).Info("Changing workflow v1 running activity")
+			ao := workflow.ActivityOptions{
+				StartToCloseTimeout: 10 * time.Second,
+			}
+			ctx1 := workflow.WithActivityOptions(ctx, ao)
+			err := workflow.ExecuteActivity(ctx1, SomeActivity, "v1").Get(ctx, nil)
+			if err != nil {
+				return err
+			}
+		} else {
+			workflow.GetLogger(ctx).Info("Concluding workflow v1", workflow.Now(ctx))
```

Comment:
Loggers expect key-value contextual data (this is missing key). Also note that most users' implementations of loggers put the time for them